File: my-app/src/main/java/com/wagonloader/app/WagonLoader.java
package com.wagonloader.app;

import java.io.IOException;
import java.util.Iterator;
import java.util.Map;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;
import com.fasterxml.jackson.databind.node.ObjectNode;
import com.fasterxml.jackson.databind.node.ValueNode;
import com.wagonloader.app.workers.WagonWorkers;

public class WagonLoader {
    ObjectMapper mapper = new ObjectMapper();
    WagonWorkers wagonWorkers = new WagonWorkers();

 

    public JsonNode fillPayload(String currentPath, JsonNode jsonNode, JsonNode wagonData) throws IOException{
        String separator = "/";
        String arraySeparator = "#";
        JsonNode result = null;

        if(jsonNode.isObject()){
            ObjectNode objectNode = (ObjectNode) jsonNode;
            Iterator<Map.Entry<String, JsonNode>> iter = objectNode.fields();
            String pathPrefix = currentPath.isEmpty() ? "" : currentPath + separator;

            ObjectNode bigNode = mapper.createObjectNode();
            
            while(iter.hasNext()){
                Map.Entry<String, JsonNode> entry = iter.next();
                result = fillPayload(pathPrefix+ entry.getKey(), entry.getValue(), wagonData);  
                bigNode.set(entry.getKey(), result);

            }
            return bigNode;
        } else if(jsonNode.isArray()){
            ArrayNode arrayNode = (ArrayNode) jsonNode;
            String pathPrefixList = currentPath.isEmpty() ? "" : currentPath + arraySeparator;

            ArrayNode bigList = mapper.createArrayNode();
            for (int i=0; i< arrayNode.size(); i++){
                result = fillPayload(pathPrefixList, arrayNode.get(i), wagonData);
                bigList.add(result);
            }
            return bigList;
        } else if(jsonNode.isValueNode()){
            ValueNode valueNode = (ValueNode) jsonNode;
            
            return wagonWorkers.evaluate(valueNode, wagonData);
            
        }

        return null;
    }

}

File: wagonloader/wagon_loader.py
import json
from wagonloader.wagon_worker_interface import WagonWorkerInterface
from wagonloader.types import JsonType

def fill_payload(node:JsonType, wagon_worker:WagonWorkerInterface) -> JsonType:

    if type(node) is dict:
        big_node = {}

        for key, value in node.items():
            result = fill_payload("", value, wagon_worker)
            big_node[key] = result
        
        return big_node

    elif type(node) is list:
        big_list = []
        for item in node:
            result_two = fill_payload("", item, wagon_worker)
            big_list.append(result_two)
        return big_list
    
    else:
        return wagon_worker.evaluate(node)
    